x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -135,8 +137,19 @@ export class RuleDataClient implements IRuleDataClient {
           },
         });
       } catch (err) {
-        // something might have created the index already, that sounds OK
-        if (err?.meta?.body?.error?.type !== 'resource_already_exists_exception') {
+        // If the index already exists and it's the write index for the alias,
```

Comment:
If the index already exists, will we ever get here? `indicesExist` will have been true in line 127. The error is silent in that case.

Reply:
It's still possible to get here if multiple code paths attempt to create the index at the same time. It can be triggered in testing if the index doesn't exist yet with a construct like 
```
ruleDataClient.getWriter().bulk(request);
ruleDataClient.getWriter().bulk(request);
```
in this case both calls race to attempt to make the index and often one hits a `resource_already_exists_exception`. It's also theoretically possible that some other source (e.g. user, backup script) could make the index in between the indices exist check and the index creation, so by fetching the index after getting the exception we can check that it was created correctly.

Reply:
True. But if the lack of a write index was the cause for this function being called we'd just fail silently because `indicesExist === true`. Would raising an exception in an `else` branch for the top level check do the trick?

Reply:
Ah I see. I was counting on the bulk retry to fail again in that case and throw the error, but I didn't notice that it wasn't checking for errors on the retry. I added handling for both in https://github.com/elastic/kibana/pull/106049/commits/28722b13558ae58ac85d42906f694af84dc0fd80 - the retry bulk call now throws errors if it encounters any, and if `createWriteTargetIfNeeded` finds existing indices then it checks to make sure one of them is the write index and throws an error otherwise. I think the bulk retry error handling alone would be sufficient to avoid silently failing, but checking the alias specifically inside `createWriteTargetIfNeeded` allows us to have a more specific error message for that case so I added both.